src/verifiers/documentStatus/tokenRegistryStatus/openAttestationEthereumTokenRegistryStatus.ts

```diff
@@ -1,22 +1,129 @@
 
-interface Status {
-  minted: boolean;
+interface ValidMintedStatus {
+  minted: true;
   address: string;
-  reason?: any;
 }
 
-const isWrappedV2Document = (document: any): document is WrappedDocument<v2.OpenAttestationDocument> => {
-  return document.data && document.data.issuers;
-};
+interface InvalidMintedStatus {
+  minted: false;
+  address: string;
+  reason: Reason;
+}
+
+type MintedStatus = ValidMintedStatus | InvalidMintedStatus;
+
 const name = "OpenAttestationEthereumTokenRegistryStatus";
 const type: VerificationFragmentType = "DOCUMENT_STATUS";
+
+export const getTokenRegistry = (
+  document: WrappedDocument<v2.OpenAttestationDocument> | WrappedDocument<v3.OpenAttestationDocument>
+): string => {
+  if (utils.isWrappedV2Document(document)) {
+    const { issuers } = getData(document);
+    if (issuers.length !== 1)
+      throw new CodedError(
+        "Only one issuer is allowed for tokens",
+        OpenAttestationEthereumTokenRegistryStatusCode.INVALID_ISSUERS,
+        OpenAttestationEthereumTokenRegistryStatusCode[OpenAttestationEthereumTokenRegistryStatusCode.INVALID_ISSUERS]
+      );
+    if (!issuers[0].tokenRegistry)
+      throw new CodedError(
+        "Token registry is undefined",
+        OpenAttestationEthereumTokenRegistryStatusCode.UNDEFINED_TOKEN_REGISTRY,
+        OpenAttestationEthereumTokenRegistryStatusCode[
+          OpenAttestationEthereumTokenRegistryStatusCode.UNDEFINED_TOKEN_REGISTRY
+        ]
+      );
+    return issuers[0].tokenRegistry;
+  } else {
+    const { proof } = getData(document);
+    if (proof.method !== "TOKEN_REGISTRY")
+      throw new CodedError(
+        "Cannot validate non-token registry documents",
+        OpenAttestationEthereumTokenRegistryStatusCode.INVALID_VALIDATION_METHOD,
+        OpenAttestationEthereumTokenRegistryStatusCode[
+          OpenAttestationEthereumTokenRegistryStatusCode.INVALID_VALIDATION_METHOD
+        ]
+      );
+    return proof.value;
+  }
+};
+
+export const isTokenMintedOnRegistry = async ({
+  tokenRegistry,
+  merkleRoot,
+  network,
+}: {
+  tokenRegistry: string;
+  merkleRoot: string;
+  network: string;
+}): Promise<MintedStatus> => {
+  try {
+    const tokenRegistryContract = await TradeTrustErc721Factory.connect(tokenRegistry, getProvider({ network }));
+    const minted = await tokenRegistryContract.ownerOf(merkleRoot).then((owner) => !(owner === constants.AddressZero));
+    return minted
+      ? { minted, address: tokenRegistry }
+      : {
+          minted,
+          address: tokenRegistry,
+          reason: {
+            code: OpenAttestationEthereumTokenRegistryStatusCode.DOCUMENT_NOT_MINTED,
+            codeString:
+              OpenAttestationEthereumTokenRegistryStatusCode[
+                OpenAttestationEthereumTokenRegistryStatusCode.DOCUMENT_NOT_MINTED
+              ],
+            message: `Document ${merkleRoot} has not been issued under contract ${tokenRegistry}`,
+          },
+        };
+  } catch (error) {
```

Comment:
maybe want to follow the same structure as in document store ? (or document store to follow the same as here)